graph/graph.go

```diff
@@ -153,10 +263,12 @@ type ConstStringNode struct {
 	FileName  string
 	LineRange Range
 	CharRange Range
+	msg       proto.Message
```

Comment:
Have you considered defining a struct that contains all of the common fields (e.g. FileName, LineRange, CharRange, msg and id) found in all the *Node structs?

Reply:
The design doc describes nodes which don't have a whistle meta data, leaving only msg and id as shared fields. I thought two shared fields didn't justify making an additional struct. Since there are only two nodes which don't use whistle meta data, an alternative could be to just set them to empty values; empty strings for the file name and zeroes for the ranges.